task.go

```diff
@@ -0,0 +1,18 @@
+package libcarina
+
+type Input interface{}
```

Comment:
Yeah, you'll want to make a struct for each permutation (though depending on the API there may be a simpler `Map` implementation). You can then conditionally decode based on the payload.